### ISSUE TITLE
Unable to set TZ as a session var unless full TZ name is used (for external table)

### ISSUE THREAD

**Issue by mcgray** [StarRocks/starrocks#38855]
Our cluster is running with UTC as global and local default. For certain external tables we would like to run queries with different TZ, say -5:00 which is EST.
We were using  https://docs.starrocks.io/docs/administration/timezone/ - as a reference. 
### Steps to reproduce the behavior (Required)
running with ` /*+ SET_VAR(time_zone = '-5:00') */`

### Expected behavior (Required)
success

### Real behavior (Required)
`SQL Error [1064] [42000]: Invalid ID for ZoneOffset, non numeric characters found: -5:00`

### Steps to reproduce the behavior (Required)
running with  `/*+ SET_VAR(time_zone = '-5') */`

### Expected behavior (Required)
success

### Real behavior (Required)
`SQL Error [1064] [42000]: can not find cctz time zone -5: file = s3://f<path>/part-00014-5f5cb9c8-0152-45dd-bb1e-95007eaec179_00009.c000.snappy.orc`

What worked for us was a full TZ name like `/*+ SET_VAR(time_zone = 'America/Detroit') */`

### StarRocks version (Required)
3.1

**Comment by packy92** [comment]
You are right. The behavior between FE and BE seems inconsistent. We will fix it ASAP.

**Comment by mcgray** [comment]
Still valid.
